dask/tests/test_order.py

```diff
@@ -653,6 +661,7 @@ def test_order_empty():
     assert order({}) == {}
 
 
+@pytest.mark.xfail(reason="Why is `cde` a better path? Why even start at a0?")
 def test_switching_dependents(abcde):
```

Comment:
I'm a little conflicted about this test case. Given this graph, the test asserts two properties

1. We start executing at `a0`
2. Once `b3` is freed we are preferring `cde` over finishing the `a*` branch

I believe 1) is up for interpretation. It is currently implemented intentionally that we start with computation branches that are "tall and narrow", i.e. we should start with `a0`. From the classical argument about starting with the longest critical path, this argument also makes sense. The critical path argument is however softened in multi threaded execution environments since `b3` is a producer of potential work since it unlocks multiple computation branches. Changing this causes other tests to fail that I'd need to investigate a little more before making an assessment on what is truly better.

Moving to 2.) assuming that we are indeed starting with `a0`, the test assertions are a little ambiguous if `a4` should be executed before `cde` or after. Regardless, there is an actual case to be made for `b3`  since it is strictly speaking the path of lower memory pressure. However, the reason why this is true is because a further splits into two tasks and assuming equal weight per task, holding onto `a7` and `a8` before we release `b3` is indeed one more task in memory.
This argument is pretty fragile and relies on how the branches split out. If any of the cde branches would further split apart, the argument would no longer be true.

So, all in all, I don't see why we would want to change our goal. Even if we were sensitive enough about the topology to infer how many final splits there are per branch (which we aren't right now and this is difficult/impossible to compute for non-trees), I'm not entirely convinced that this is reason alone to "switch". I could see a stronger case about changing the starting point, though.

Given that there is already a disclaimer that this is specific to the current implementation, I'm inclined to skip this for now and move on to testing. If I decide to go through with this implementation, I will likely delete the test case.